python/grass/jupyter/interact_display.py

```diff
@@ -0,0 +1,158 @@
+        else:
+            rcfile_location = os.environ["GISRC"]
+            
+        with open(rcfile_location) as f:
+            for line in f:
+                entry = line.strip("\n").split(": ")
+                rc[entry[0]] = entry[1]
+        return rc
+    
+    
+
+    def _convert_coordinates(self, x, y, proj_in):
+        """This function reprojects coordinates to WGS84, the required
+        projection for folium.
+
+        Coordinates should be a string with x and y values separated
+        by a comma.
+
+        proj_in is a proj4 string, for example, the output of g.region
+        with the `g` flag."""
+        # Reformat input
+        coordinates = f"{x}, {y}"
+        
+        # Reproject coordinates
+        coords_folium = gs.read_command(
+            "m.proj",
+            coordinates=coordinates,
+            proj_in=proj_in,
+            separator="comma",
+            flags="do",
+        )
+
+        # Reformat from string to array
+        coords_folium = coords_folium.strip()  # Remove '\n' at end of string
+        coords_folium = coords_folium.split(",")  # Split on comma
+        coords_folium = [float(value) for value in coords_folium]  # Convert to floats
+
+        return coords_folium[1], coords_folium[0]  # Return Lat and Lon
+
+    def _convert_extent(self, env=None):
+        """This function returns the bounding box of the current region
+        in WGS84, the required projection for folium"""
+
+        # Get proj of current GRASS region
+        proj = gs.read_command("g.proj", flags="jf", env=env)
+
+        # Get extent
+        extent = gs.parse_command("g.region", flags="g", env=env)
+
+        # Convert extent to EPSG:3857, required projection for Folium
+        north, east = self._convert_coordinates(extent["e"], extent["n"], proj)
+        south, west = self._convert_coordinates(extent["w"], extent["s"], proj)
+
+        extent = {'north': north, 'south': south, 'east': east, 'west': west}
+
+        return extent
+    
+    def _folium_bounding_box(self, extent):
+        """Reformats extent into bounding box to pass to folium"""
+        return [[extent['north'], extent['west']], [extent['south'], extent['east']]]
+    
+    def add_vector(self, vector):
+        """Imports vector into temporary WGS84 location, 
+        re-formats to a GeoJSON and adds to folium map"""
+        
+        # Import vector into new Location/Mapset
+        name = gs.find_file(vector, element="vector")["fullname"]
+        # Reproject vector into WGS84 Location
+        gs.run_command("v.proj",
+                       input=name,
+                       location=self._rc_original["LOCATION_NAME"],
+                       mapset=self._rc_original["MAPSET"],
+                       env=self._env
+                      )
+        
+        # Convert to GeoJSON
+        json_file = f"tmp_{vector}.json"
+
+        gs.run_command("v.out.ogr", input=name, output=json_file, format="GeoJSON", env=self._env)
```

Comment:
This works I think only because the vector happens to be in the PERMANENT mapset in the original/current location. This may or may not be the case generally. In the `self._env` location, it will always be in PERMANENT. Specifying only the name without mapset seems to me like the best way to go. You just reprojected that to whatever is the current mapset for `self._env`, so specifying the mapset is not needed.

To cover all the cases, you need to really have the original name here, but perhaps stripped from the mapset - right, you should support things like `add_vector("streams@user1")`. Fortunately, just passing that to _find_file_ should work (I just tested with _g.findfile_). It will resolve all the pieces for you. In other words, passing the fullname is wrong, passing the original parameter value won't work for `name@mapset` case, use full _find_file_ result to reliably get the pieces you need.

Reply:
I think this is resolved?